scripts/classificationloss.py

```diff
@@ -0,0 +1,220 @@
+            results[clid].label = results[clid].label - 1
+
+        return results
+
+    @staticmethod
+    def calcDataset(dataset, model, iou_thresh=0.5, verbose=True, progress=True):
+        # Perform performance analysis on dataset
+        boxes_pred = []
+        boxes_gt = []
+
+        num_images = len(dataset.images)
+
+        for imnr in range(num_images):
+            if progress:
+                printProgressBar(
+                    imnr, num_images, prefix="Progresss:", suffix="Complete", length=50
+                )
+
+            imgpath = dataset.images[imnr]
+            rects = dataset.rects[imnr]
+            im = Image.open(imgpath)
+            im_class_nr = [r["classid"] for r in rects]
+            im_class = [dataset.classes[r["classid"]] for r in rects]
+            im_roi = [r["rect"] for r in rects]
+
+            frame = np.array(im, dtype=model.dtype)
+            class_ids, confidences, boxes = model.apply_model(frame)
+            for class_id, confidence, box in zip(class_ids, confidences, boxes):
+                box_pred = box_xywh_to_xyxy(box)
+                label_pred = class_id + 1
+                conf_pred = confidence
+                boxes_pred.append(
+                    metrics.BoundingBox.of_bbox(
+                        imnr,
+                        label_pred,
+                        box_pred[0],
+                        box_pred[1],
+                        box_pred[2],
+                        box_pred[3],
+                        conf_pred,
+                    )
+                )
+
+            for box, label in zip(im_roi, im_class_nr):
+                box_gt = box_xywh_to_xyxy(box)
+                label_gt = label + 1
+                boxes_gt.append(
+                    metrics.BoundingBox.of_bbox(
+                        imnr, label_gt, box_gt[0], box_gt[1], box_gt[2], box_gt[3], 1.0
+                    )
+                )
+
+        results = metrics.get_pascal_voc_metrics(boxes_gt, boxes_pred, iou_thresh)
+
+        # remove class offset, offset added as some algorithms ingore class-0 as background
+        classes = list(results.keys())
+        for clid in classes:
+            results[clid].label = results[clid].label - 1
+
+        # Print results if requested
+        if verbose:
+            for cls, metric in results.items():
+                label = metric.label
+                print("class", dataset.classes[label], " (" + str(label) + ")")
+                print("ap", metric.ap)
+                print("precision", metric.precision)
+                print("interpolated_recall", metric.interpolated_recall)
+                print("interpolated_precision", metric.interpolated_precision)
+                print("tp", metric.tp)
+                print("fp", metric.fp)
+                print("num_groundtruth", metric.num_groundtruth)
+                print("num_detection", metric.num_detection)
+                print("--------------------")
+
+        return results
```

Comment:
Again, I'm really not seeing the point of wrapping this and doing all the conversions when we can interface with podm directly with the COCO dataset type given by roboflow.